packages/store/src/utils/disposable.ts

```diff
@@ -2,17 +2,51 @@ export interface Disposable {
   dispose(): void;
 }
 
-export function flattenDisposable(a: Disposable[]): Disposable {
+type DisposeLogic = Disposable | (() => void);
+
+export class DisposableGroup implements Disposable {
```

Comment:
Does the `DisposableGroup` designed to automatically dispose all the disposable objected added to it? This looks good!

Reply:
Yes, exactly!